quil-rs/src/program/graph.rs

```diff
@@ -474,29 +476,29 @@ impl InstructionBlock {
         self.instructions.is_empty()
     }
 
-    pub fn set_exit_condition(&mut self, terminator: BlockTerminator) {
-        self.terminator = terminator
+    pub fn set_exit_condition(&mut self, terminator: BlockTerminator<'a>) {
+        self.terminator = terminator;
     }
 }
 
 #[derive(Clone, Debug)]
-pub enum BlockTerminator {
+pub enum BlockTerminator<'a> {
     Conditional {
-        condition: MemoryReference,
-        target: String,
+        condition: &'a MemoryReference,
+        target: &'a String,
```

Comment:
If you're including lifetimes now, does it make sense/help at all to move from `&'a String` to `&'a str`?